sagemaker_neo_compilation_jobs/segmentation_fcn_resnet50_ade_neo/segmentation_fcn_resnet50_ade_neo.ipynb

```diff
@@ -0,0 +1,328 @@
+    "model_path = sess.upload_data(path='fcn_resnet50_ade.tar.gz', key_prefix=prefix)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load model using `sagemaker.MXNetModel`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.mxnet.model import MXNetModel\n",
+    "from sagemaker.predictor import RealTimePredictor\n",
+    "\n",
+    "segmentation_model = MXNetModel(model_data=model_path,\n",
+    "                                image=image_url,\n",
+    "                                entry_point='entry_point.py',\n",
+    "                                predictor_cls=RealTimePredictor,\n",
+    "                                role=role,\n",
+    "                                sagemaker_session=sess,\n",
+    "                                py_version='py3',\n",
+    "                                framework_version=framework_version)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compile model for `ml_p3` target using Neo. \n",
+    "\n",
+    "After compilation, we will deploy Neo optimized model to do inderence. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "compilation_job_name = name_from_base('Segmentation-fcn-resnet50-Neo')\n",
+    "output_path = 's3://{}/{}'.format(bucket, prefix)\n",
+    "compiled_model = segmentation_model.compile(target_instance_family='ml_p3', \n",
+    "                                            input_shape={'data':[1,3,480,480]},\n",
+    "                                            job_name=compilation_job_name,\n",
+    "                                            role=role,\n",
+    "                                            framework=framework,\n",
+    "                                            output_path=output_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Inference\n",
+    "\n",
+    "Now that the model is deployed at an endpoint that is up-and-running, we can use this endpoint for inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "object_detector = compiled_model.deploy(initial_instance_count = 1,\n",
+    "                                        instance_type = 'ml.p3.2xlarge')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# download the example image\n",
+    "!wget -O test.jpg https://github.com/zhanghang1989/image-data/blob/master/encoding/segmentation/ade20k/ADE_val_00001755.jpg?raw=true\n",
+    "\n",
+    "file_name = \"test.jpg\"\n",
+    "\n",
+    "with open(file_name, 'rb') as f:\n",
+    "    payload = f.read()\n",
+    "    payload = bytearray(payload)"
+   ]
+  },
+  {
```

Comment:
ModelError: An error occurred (ModelError) when calling the InvokeEndpoint operation: Received server error (0) from model with message "Your invocation timed out while waiting for a response from container model."

What notebook instance type did you use?

Reply:
The notebook instance type I used is `ml.m4.2xlarge`. I think this notebook is outdated. I published this PR months ago, not sure we still want to merge this.

Reply:
Noted. 

Will it benefit the customer to have this notebook? As far as I can tell, we don't have examples of segmentation model so it might be useful to keep.

If yay, can you update it so it works?

If nay, let's close this PR.